src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    public static final int RADIUS_OF_CIRCLE = 10;
+    public static final int MAX_BOUND = 100;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+
+        switch (random.nextInt(FIGURE_COUNT)) {
+            case 0:
+                return new Square(random.nextInt(MAX_BOUND), colorSupplier.getRandomColor());
```

Comment:
According the common mistakes "Don't pass expressions to the constructor." Please introduce a appropriate variables for the expressions (`random.nextInt(MAX_BOUND)`, `colorSupplier.getRandomColor()`). Fix here and in similar cases in the code below

Reply:
Not fixed. See next comment for the example.